internal/api/impl/auth/oauth.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+	"github.com/perses/perses/internal/api/shared"
+	"github.com/perses/perses/internal/api/shared/crypto"
+	"github.com/perses/perses/internal/api/shared/route"
+	"github.com/perses/perses/pkg/model/api/config"
+	"golang.org/x/oauth2"
+)
+
+type oAuthEndpoint struct {
+	conf            *oauth2.Config
+	jwt             crypto.JWT
+	tokenManagement tokenManagement
+	slugID          string
+	userInfoURL     string
+}
+
+func newOAuthEndpoint(params config.OAuthProvider, jwt crypto.JWT) authEndpoint {
+	conf := &oauth2.Config{
+		ClientID:     string(params.ClientID),
+		ClientSecret: string(params.ClientSecret),
+		Scopes:       params.Scopes,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  params.AuthURL,
+			TokenURL: params.TokenURL,
+		},
+		RedirectURL: params.RedirectURI,
+	}
+
+	return &oAuthEndpoint{
+		conf:            conf,
+		jwt:             jwt,
+		tokenManagement: tokenManagement{jwt: jwt},
+		slugID:          params.SlugID,
+		userInfoURL:     params.UserInfosURL,
+	}
+}
+
+func (e *oAuthEndpoint) CollectRoutes(g *route.Group) {
+	g.GET(fmt.Sprintf("/%s/%s", e.String(), loginPath), e.authHandler, true)
```

Comment:
same suggestion than for OIDC

```suggestion
g.GET(fmt.Sprintf("/%s/%s", e.String(), loginPath), e.authHandler, false)
```

the login endpoint shouldn't require a JWT token to use it I think

Reply:
nono the login endpoint will trigger the authorization code flow that will redirect to the callback (code exchange) endpoint that will generate the token. so tehse two ednpoints should be anonymous